apps/backend/app/services/world_entity_service.py

```diff
@@ -0,0 +1,262 @@
+        known = get_entity_names_by_story_id(self.db_session, self.story.id)
+        return [name for name in entity_names if name.lower() not in {k.lower() for k in known}]
+
+    async def describe_entity(self, entity_name: str, scene_text: str, related_entities: List[Dict]) -> Optional[Dict[str, Any]]:
+        """
+        Generate a canonical description for a single world entity using scene and world context.
+        """
+        if not self.story:
+            logging.warning("No story context available for entity description")
+            return None
+
+        related_str = "\n".join([f"{e['name']}: {e['description']}" for e in related_entities])
+
+        system_prompt = f"""
+        > You are building a glossary entry for the term "{entity_name}" in a fictional game world.
+
+        > This game world has its own unique lore, characters, and settings. The term "{entity_name}" is a concept or term that characters in the world would know about.
+
+        > Here is the description of the world/story:
+        {self.story.description}
+
+        > The term was encountered in this scene:
+        ---
+        {scene_text}
+        ---
+
+        > Here are known, related world concepts:
+        {related_str if related_entities else 'None'}
+
+        > Task:
+        - Write a short canonical description (1–3 sentences) that could be added to a world glossary.
+        - Be concise, specific, and avoid repeating known concepts.
+        - Assume the reader is a character in the world who already knows general context.
+        - The term must be defined in a way that is useful for all characters in the world.
+        - The definition must only include information that would be common knowledge to characters in the world.
+        - Especially, it must avoid referring to any characters or situations that are not famous or well-known in the world.
+        - As this description will become canonical, feel free to use your own creativity to fill in the gaps where information is not present.
+        - You are co-authoring the world with the user, so feel free to add your own creative flair.
+        - Keep the style of the description consistent with the world and its lore.
+        - Make the format encyclopedic, as if it were a Wikipedia entry - but brief and to the point.
+
+        > Output format:
+        {{
+            "name": "{entity_name}",
+            "description": "Your generated description here."
+        }}
+        """
+
+        messages = [
+            self.llm_service.create_message("system", system_prompt)
+        ]
+
+        response = await self.llm_service.generate_completion(
+            messages=messages,
+            model=ModelName.GEMINI_2_FLASH_LITE,
+            temperature=0.4,
+            stream=False
+        )
+
+        content = await self.llm_service.extract_content(response)
+        return JSONService.parse_and_validate_single_object(content, required_keys=["name", "description"])
```

Comment:
create the schema WorldEntityFromLLM (or similar) that will have name and description and use `parse_and_validate_json_response` as in other places